Index: src/main/java/com/example/clashofclans/Model/Building/Powder.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package com.example.clashofclans.Model.Building;\r\n\r\nimport com.example.clashofclans.Values;\r\nimport javafx.scene.image.Image;\r\n\r\npublic class Powder extends DefensiveBuilding{\r\n    public Powder() {\r\n        super(Values.POWDER_DAMAGE, Values.POWDER_HIT, Values.POWDER_HITRANGE);\r\n        this.setImagePath(getClass().getResource(\"Building/Powder.png\").toString());\r\n    }\r\n}\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/com/example/clashofclans/Model/Building/Powder.java b/src/main/java/com/example/clashofclans/Model/Building/Powder.java
--- a/src/main/java/com/example/clashofclans/Model/Building/Powder.java	(revision 2e2195876d51d31efe4ee23a844119fcfdf2b8f9)
+++ b/src/main/java/com/example/clashofclans/Model/Building/Powder.java	(date 1687899598151)
@@ -1,11 +1,11 @@
 package com.example.clashofclans.Model.Building;
 
+import com.example.clashofclans.HelloApplication;
 import com.example.clashofclans.Values;
 import javafx.scene.image.Image;
 
 public class Powder extends DefensiveBuilding{
     public Powder() {
-        super(Values.POWDER_DAMAGE, Values.POWDER_HIT, Values.POWDER_HITRANGE);
-        this.setImagePath(getClass().getResource("Building/Powder.png").toString());
+        super(Values.POWDER_DAMAGE, Values.POWDER_HIT, Values.POWDER_HITRANGE,new HelloApplication().getClass().getResource("Building/Powder").toString());
     }
 }
Index: src/main/java/com/example/clashofclans/Model/Building/Thrower.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package com.example.clashofclans.Model.Building;\r\n\r\nimport com.example.clashofclans.Values;\r\nimport javafx.scene.image.Image;\r\n\r\npublic class Thrower extends DefensiveBuilding {\r\n    public Thrower() {\r\n        super(Values.THROWER_DAMAGE, Values.THROWER_HIT, Values.THROWER_HITRANGE);\r\n        this.setImagePath(getClass().getResource(\"Building/Thrower.png\").toString());\r\n    }\r\n}\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/com/example/clashofclans/Model/Building/Thrower.java b/src/main/java/com/example/clashofclans/Model/Building/Thrower.java
--- a/src/main/java/com/example/clashofclans/Model/Building/Thrower.java	(revision 2e2195876d51d31efe4ee23a844119fcfdf2b8f9)
+++ b/src/main/java/com/example/clashofclans/Model/Building/Thrower.java	(date 1687899598160)
@@ -1,11 +1,11 @@
 package com.example.clashofclans.Model.Building;
 
+import com.example.clashofclans.HelloApplication;
 import com.example.clashofclans.Values;
 import javafx.scene.image.Image;
 
 public class Thrower extends DefensiveBuilding {
     public Thrower() {
-        super(Values.THROWER_DAMAGE, Values.THROWER_HIT, Values.THROWER_HITRANGE);
-        this.setImagePath(getClass().getResource("Building/Thrower.png").toString());
+        super(Values.THROWER_DAMAGE, Values.THROWER_HIT, Values.THROWER_HITRANGE,new HelloApplication().getClass().getResource("Building/Thrower.png").toString());
     }
 }
Index: src/main/java/com/example/clashofclans/Model/Building/Canon.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package com.example.clashofclans.Model.Building;\r\n\r\nimport com.example.clashofclans.Values;\r\nimport javafx.scene.image.Image;\r\n\r\npublic class Canon extends DefensiveBuilding {\r\n\r\n    public Canon() {\r\n        super(Values.CANON_DAMAGE, Values.CANON_HIT, Values.CANON_HITRANGE);\r\n        this.setImagePath(null);\r\n    }\r\n\r\n    @Override\r\n    public void setImagePath(String imagePath){\r\n        super.setImagePath(getClass().getResource(\"Heroes/Knight/Knight_01__WALK_006.png\").toString());\r\n    }\r\n\r\n}\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/com/example/clashofclans/Model/Building/Canon.java b/src/main/java/com/example/clashofclans/Model/Building/Canon.java
--- a/src/main/java/com/example/clashofclans/Model/Building/Canon.java	(revision 2e2195876d51d31efe4ee23a844119fcfdf2b8f9)
+++ b/src/main/java/com/example/clashofclans/Model/Building/Canon.java	(date 1687899635727)
@@ -1,18 +1,13 @@
 package com.example.clashofclans.Model.Building;
 
+import com.example.clashofclans.HelloApplication;
 import com.example.clashofclans.Values;
 import javafx.scene.image.Image;
 
 public class Canon extends DefensiveBuilding {
 
     public Canon() {
-        super(Values.CANON_DAMAGE, Values.CANON_HIT, Values.CANON_HITRANGE);
-        this.setImagePath(null);
-    }
-
-    @Override
-    public void setImagePath(String imagePath){
-        super.setImagePath(getClass().getResource("Heroes/Knight/Knight_01__WALK_006.png").toString());
+        super(Values.CANON_DAMAGE, Values.CANON_HIT, Values.CANON_HITRANGE,new HelloApplication().getClass().getResource("Building/Canon.png").toString());
     }
 
 }
Index: src/main/java/com/example/clashofclans/Model/Building/Building.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package com.example.clashofclans.Model.Building;\r\n\r\nimport com.example.clashofclans.Values;\r\nimport javafx.scene.image.Image;\r\n\r\nimport java.io.FileInputStream;\r\n\r\npublic class Building {\r\n    private int damage;\r\n    private Values.DefensiveType type;\r\n    private String imagePath;\r\n\r\n    public Building(int damage, Values.DefensiveType type) {\r\n        this.damage = damage;\r\n        this.type = type;\r\n    }\r\n\r\n    public int getDamage() {\r\n        return damage;\r\n    }\r\n\r\n    public void setDamage(int damage) {\r\n        this.damage = damage;\r\n    }\r\n\r\n    public Values.DefensiveType getType() {\r\n        return type;\r\n    }\r\n\r\n    public void setType(Values.DefensiveType type) {\r\n        this.type = type;\r\n    }\r\n\r\n    public String getImagePath() {\r\n        return imagePath;\r\n    }\r\n\r\n    public void setImagePath(String imagePath) {\r\n        this.imagePath = imagePath;\r\n    }\r\n}\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/com/example/clashofclans/Model/Building/Building.java b/src/main/java/com/example/clashofclans/Model/Building/Building.java
--- a/src/main/java/com/example/clashofclans/Model/Building/Building.java	(revision 2e2195876d51d31efe4ee23a844119fcfdf2b8f9)
+++ b/src/main/java/com/example/clashofclans/Model/Building/Building.java	(date 1687899450479)
@@ -5,12 +5,12 @@
 
 import java.io.FileInputStream;
 
-public class Building {
+public class Building extends Image{
     private int damage;
     private Values.DefensiveType type;
-    private String imagePath;
 
-    public Building(int damage, Values.DefensiveType type) {
+    public Building(int damage, Values.DefensiveType type,String imagePath) {
+        super(imagePath);
         this.damage = damage;
         this.type = type;
     }
@@ -31,11 +31,4 @@
         this.type = type;
     }
 
-    public String getImagePath() {
-        return imagePath;
-    }
-
-    public void setImagePath(String imagePath) {
-        this.imagePath = imagePath;
-    }
 }
Index: src/main/java/com/example/clashofclans/Model/Building/DefensiveBuilding.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package com.example.clashofclans.Model.Building;\r\n\r\nimport com.example.clashofclans.Values;\r\n\r\npublic class DefensiveBuilding extends Building {\r\n    private int hit;\r\n    private int hitRange;\r\n\r\n    public DefensiveBuilding(int damage, int hit, int hitRange) {\r\n        super(damage, Values.DefensiveType.DEFENSIVE);\r\n        this.hit = hit;\r\n        this.hitRange = hitRange;\r\n    }\r\n\r\n    public int getHit() {\r\n        return hit;\r\n    }\r\n\r\n    public void setHit(int hit) {\r\n        this.hit = hit;\r\n    }\r\n\r\n    public int getHitRange() {\r\n        return hitRange;\r\n    }\r\n\r\n    public void setHitRange(int hitRange) {\r\n        this.hitRange = hitRange;\r\n    }\r\n}\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/com/example/clashofclans/Model/Building/DefensiveBuilding.java b/src/main/java/com/example/clashofclans/Model/Building/DefensiveBuilding.java
--- a/src/main/java/com/example/clashofclans/Model/Building/DefensiveBuilding.java	(revision 2e2195876d51d31efe4ee23a844119fcfdf2b8f9)
+++ b/src/main/java/com/example/clashofclans/Model/Building/DefensiveBuilding.java	(date 1687899598146)
@@ -6,8 +6,8 @@
     private int hit;
     private int hitRange;
 
-    public DefensiveBuilding(int damage, int hit, int hitRange) {
-        super(damage, Values.DefensiveType.DEFENSIVE);
+    public DefensiveBuilding(int damage, int hit, int hitRange,String imagePath) {
+        super(damage, Values.DefensiveType.DEFENSIVE,imagePath);
         this.hit = hit;
         this.hitRange = hitRange;
     }
